### DISCUSSION TITLE
Malt not initializing when being used in other scenes.

### DISCUSSION THREAD

**Discussion by aVersionOfReality** [bnpr/Malt#199] [General]
I'm not sure if I should be reporting this as a bug, or if there's something that can be done about it.

I have a scene that uses Cycles mainly, but then has 2 secondary Malt scenes for different layers of Line Art. All of them are composited together in the Cycles scene layer. The problem is that if I open the file and just hit f12 render, the Malt layers render black and the Malt window never opens. I have to go into the other scenes and start a viewport render in each. Then I can flip back to the Cycles scene and do the full render.

Is there a way to fix this? I am using the stable build from Dec10.

**Comment by pragma37** [comment]
I think this should be fixed in the latest release.
[multi-render.zip](https://github.com/bnpr/Malt/files/7705200/multi-render.zip)

The render view only shows the Malt render for some reason, but if you save the render or swap the slots back and forth you get the composited result.
Seems like a Blender bug?

**Reply by aVersionOfReality** [reply]
Great. This is working, but still note entirely. It works when the material is the default, or set as a material on the object. But if you are using an Override material (which I am in the scene I'm working with), then the render just comes out black still. This happens for me in both my big scene, and in the multi-render file you linked here.

**Reply by pragma37** [reply]
Did you maybe swapped from Release to Development?

You need to let Malt initialice the scene at least once.
Then, as long as you have saved, you should be able to render directly from the Cycles scene the next time you open it.

This one works for me on the Development branch:
[multi-render.zip](https://github.com/bnpr/Malt/files/7722134/multi-render.zip)

If you still find problems upload a minimal reproducible example and tell me what version you are using.

**Reply by aVersionOfReality** [reply]
Alright, I am using today's Development Build (Dec15) and the multi-render file you linked (and Blender 3.0). I've changed the UI to show the compositor and render, and plugged the malt scene into the output node so I can see if it rendered.
Here's the behavior I'm getting:

1: Open the File and immediately hit F12. A Malt window pops up. The Eevee render layer runs. The Malt scene runs but returns only Alpha. I get this in the console:
![image](https://user-images.githubusercontent.com/36803956/146291422-f31d52dc-3ac8-46b6-8655-bf9cdfafe302.png)

2: Open the File and switch to the Malt scene. The console continuously spams this: 
![image](https://user-images.githubusercontent.com/36803956/146291619-f58dce64-05f6-4ec0-8884-166018c5af60.png)
Attempting to viewport render gives only Alpha.
When I select the Malt render engine, a Malt window pops up and the console stops spamming. Viewport rendering now works. If I switch back to the Eevee scene and F12 render, it all comes out properly.

3: I saved the file after 2, reopened it, and immediately hit F12. The result was the same as 1.

4: As 2, but before switching back to the Eevee scene I saved and re-opened the file. Malt pops up as the file opens. Viewport rendering works right away. Switching to the Eevee scene and F12 rendering works fine.

5: Open the file into the Eevee scene as in 1. Mute and then Unmute the Malt Renderlayer node in the compositor (or re-select the scene). Malt pops up a window when its muted, and again when it is unmuted. F12 render then pops up a Malt window a 3rd time, and works without issue.

So it does work once Malt has been initialized, but does not continue to work after saving and reopening. Malt will initialize on file open when opening onto a Malt scene, but does not initialize when switching to a Malt scene.

Is there any simple or proper way to call Malt to initialize? I could use a script that runs at file open or something so that it'll come up even in non-malt scenes, or on Scene changes. I guess I could have the script just toggle the Render engine to do it?

6: I added a second Malt scene (my real scene file uses 2, although I may be able to switch to 2 renderlayers in 1 scene) by doing a full copy of the first, and adding another Object so I can tell them apart on renders. I also added a render layer node for it to the compositor in the Eevee scene.
If I open the file on the Eevee scene and initialize only Malt 1 by re-selecting it in the render layer and then F12 render, it works but the Compositor doesn't update the previews for the renderlayer nodes until I change something to make it run again.
If I open the file and initialize Malt 1 and then Malt 2 and then F12 Render, it does render but then the Malt 1 node's output sockets disappear and I get this error:

![image](https://user-images.githubusercontent.com/36803956/146294385-3a5872ad-5d39-4f3b-a582-e82eeb55530f.png)

If I make a compositor change like plugging Malt 2 into the output, then the compositor refreshes, Malt initializes again. The sockets reappear and the render previews appear on the nodes. So there's some sort of conflict between the two Scenes.

Thanks for the help!
